### PR TITLE
Cwdoe 1384 inject types

### PR DESCRIPTION
**Pull request by sei-tspencer**
DataField template revisions

### CODE REVIEW

File: src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts [hunk 1]
```diff
@@ -67,20 +64,18 @@ export class DataFieldEditDialogComponent {
...
         const defaultOptions = this.settingsService.settings.DefaultDeliveryMethodOptions;
         defaultOptions.forEach(defopt => {
-          const alreadyExists = this.data.dataFieldOptions.some(dfo =>
+          const alreadyExists = this.data.dataField.dataOptions.some(dfo =>
             dfo.optionName === defopt.optName && dfo.optionValue === defopt.optionValue);
           if (!alreadyExists) {
             const newDataOption = {
               id: uuidv4(),
-              displayOrder: this.data.dataFieldOptions.length + 1,
+              displayOrder: this.data.dataField.dataOptions.length + 1,
```
Comment (sei-mpriest) [inline src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts:72]: It looks like the Reorder method in DataFieldService expects the displayOrder to be zero-based and this expects it to be one-based. Note: That Reorder is a bit strange.

File: src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts [hunk 2]
```diff
@@ -91,7 +86,7 @@ export class DataFieldEditDialogComponent {
 
   addDataOption(dataField: DataField) {
     this.addOrEditDataOption({
-      displayOrder: this.data.dataFieldOptions.length + 1,
+      displayOrder: this.data.dataField.dataOptions.length + 1,
```
Comment (sei-mpriest) [inline src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts:89]: We should probably make it one greater than the current greatest value.

File: src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts [hunk 3]
```diff
@@ -113,32 +108,13 @@ export class DataFieldEditDialogComponent {
...
+          const existingIndex = this.data.dataField.dataOptions.findIndex(x => x.id === dataOption.id);
+          this.data.dataField.dataOptions[existingIndex] = dataOption;
         } else {
           // no id means this data option has just been entered
           dataOption.id = uuidv4();
           // add to displayed data options
-          this.data.dataFieldOptions.push(dataOption);
-          // make sure data option gets added when we are done
-          this.addedDataFieldOptions.push(dataOption);
+          this.data.dataField.dataOptions.push(dataOption);
```
Comment (sei-mpriest) [inline src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts:117]: Do we have to worry about displayOrder here?

File: src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts (hunk 1 above)
Comment (sei-tspencer) [reply src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts:72]: The reorder in DataFieldService reorders DataFields.  This displayOrder is for the DataOptions contained in a DataField.  Two different lists.  We can talk through the DataFieldService Reorder method if you would like to.

File: src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts (hunk 2 above)
Comment (sei-tspencer) [reply src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts:89]: This is just the value that gets pre-populated into the edit modal.  The user can correct it if it needs adjusting.  We currently do nothing to reorder DataOptions automatically.  Go ahead and add a Jira ticket to add that functionality.

File: src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts (hunk 3 above)
Comment (sei-tspencer) [reply src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts:117]: Per my comment above, go ahead and add a Jira ticket for that.

File: src/app/components/data-field-list/data-field-list.component.ts [hunk 4]
```diff
@@ -151,20 +168,24 @@ export class DataFieldListComponent implements OnDestroy, OnInit {
       if (makeTemplate) {
         dataField.mselId = null;
         dataField.injectTypeId = null;
+        dataField.displayOrder = 99;
```
Comment (sei-mpriest) [inline src/app/components/data-field-list/data-field-list.component.ts:171]: Maybe just leave it empty?

Comment (sei-tspencer) [reply src/app/components/data-field-list/data-field-list.component.ts]: change pushed

**Review by sei-mpriest — APPROVED** [review]

**Pull request merged**